format.c

```diff
@@ -370,6 +377,28 @@ static int unpack (FILE *fp, const char *what, const char *fmt,
 }
 
 
+/**
+ * Convert timestamp into a human-readable format.
+ */
+void format_timestamp(char *dst, size_t maxsize, int64_t unixtime_ms) {
+    time_t unixtime_secs = unixtime_ms / 1000;
+
+    struct tm *ts = localtime(&unixtime_secs);
+
+    size_t pos = strftime(dst, maxsize,
+                          "%Y-%m-%dT%H:%M:%S.", ts);
+
+    // add milliseconds
+    pos += snprintf(
+            &dst[pos],
+            maxsize-pos,
+            "%" PRId64,
+            unixtime_ms % 1000);
+
```

Comment:
pos may be > maxsize-pos here if we ran out of space, so only append the timezone if pos < maxsize

Reply:
Ooops. I've added a check here and after strftime.